src/asynchronous/client.rs

```diff
@@ -117,13 +117,20 @@ impl Client {
         streaming_server: bool,
     ) -> Result<StreamInner> {
         let stream_id = self.next_stream_id.fetch_add(2, Ordering::Relaxed);
+        let is_req_payload_empty = req.payload.is_empty();
```

Comment:
nit: This looks like it is only used when `stream_client` is enabled.  Could move to the line 127?

Reply:
unfortunately the compiler says: 
```
error[E0382]: borrow of moved value: `req`
   --> src/asynchronous/client.rs:126:17
    |
115 |         req: Request,
    |         --- move occurs because `req` has type `ttrpc::Request`, which does not implement the `Copy` trait
...
121 |         let mut msg: GenMessage = Message::new_request(stream_id, req)?
    |                                                                   --- value moved here
...
126 |             if !req.payload.is_empty() {
    |                 ^^^^^^^^^^^^^^^^^^^^^^ value borrowed here after move
```